src/hooks/useLocations.js

```diff
@@ -15,30 +16,48 @@ export const useLocations = (lat, lng) => {
     const baseURL = 'https://segdeha.com/api/nearby.php';
     const completeURL = lat && lng && `${baseURL}?lat=${lat}&lng=${lng}`;
 
-    setStatus('loading');
-    fetch(completeURL, { signal })
-      .then((response) => {
-        setIsFetching(true);
-        return response.json();
-      })
-      .then((data) => {
-        setLocations(data?.query?.pages);
-        setStatus('success');
-      })
-      .catch((err) => {
-        if (err.name !== 'AbortError') {
-          setStatus('error');
-          setError(err);
-        }
-      })
-      .finally(() => setIsFetching(false));
-
-    // Clean up
-    return function cancel() {
-      setStatus('cancelled');
-      abortController?.abort();
+    const fetchLocations = () => {
+      setStatus('loading');
+      fetch(completeURL, { signal })
+        .then((response) => {
+          setIsFetching(true);
+          return response.json();
+        })
+        .then((data) => {
+          setLocations(data?.query?.pages);
+          setStatus('success');
+        })
+        .catch((err) => {
+          if (err.name !== 'AbortError') {
+            setStatus('error');
+            setError(err);
+          }
+        })
+        .finally(() => setIsFetching(false));
+
+      //Clean up
+      return function cancel() {
+        setStatus('cancelled');
+        abortController?.abort();
+      };
     };
-  }, [lat, lng]);
+
+    if (isDragged) {
+      fetchLocations();
+      return;
+    }
+
+    if (Date.now() >= timeRef.current + interval) {
+      timeRef.current = Date.now();
+      fetchLocations();
+    } else {
+      const id = setTimeout(() => {
+        timeRef.current = Date.now();
+        fetchLocations();
+      }, interval - (Date.now() - timeRef.current));
+      return () => clearTimeout(id);
+    }
```

Comment:
Took me a second to understand what you did here, but in the end I like it! 🙌

Reply:
I'm honest. I'm so sorry because I couldn't understand it completely but I guess I need to read more about `setTimeout` and `on-the-fly` math operations also may be in the demo.